app/views/samples/_form_js.haml

```diff
@@ -1,13 +1,22 @@
 :javascript
   (function() {
-    var onloadPicture = function(filename) {
+    var onloadPicture = function(file) {
       return function(event) {
-        var fileDiv = $('<div>', { class: 'picture-container new-file' })
-          .append($('<img>', { class: 'uploaded', src: event.target.result }))
-          .append($('<div>', { class: 'picture-title' }).text(filename));
+        var src = "";
+        var filetype = file.type;
+        if (filetype == 'image/png' || filetype == 'image/jpeg' || filetype == 'image/gif') {
+          src = event.target.result;
+        }
 
-        $('.assay > .upload-new-file')
-          .append(fileDiv);
+        $('.assay > .upload-new-file').append(
+          $('<div>', { class: 'picture-container new-file assay-file' })
+            .append((src == "") ? $('<div>', { class: 'any-filetype icon-document' }) : $('<img>', { class: 'uploaded', src: src }))
```

Comment:
The usage of the `src` variable here is weird. You can abstract this to make it more readable. Something like this:
```js
function fileThumbnail(file, url) {
  if (file.type.startsWith('image/')) {
    return $('<img>', { class: 'uploaded', src: url });
  } else {
    return $('<div>', { class: 'any-filetype icon-document' });
  }
}
```